src/goto-symex/symex_function_call.cpp

```diff
@@ -431,28 +427,20 @@ void goto_symext::return_assignment(statet &state)
 
   target.location(state.guard.as_expr(), state.source);
 
-  if(code.operands().size()==1)
-  {
-    exprt value=code.op0();
+  PRECONDITION(code.operands().size() == 1);
```

Comment:
This is not consistent with the previous condition - this should be `code.operands.size() == 1 || frame.return_value.is_nil()` - as is, this would now incorrectly fail with returns that don't have a parameter.